### ISSUE TITLE
Error from grib_get_message() with certain files

### ISSUE THREAD

**Issue by anchiho** [nawendt/gribr#21]
Hello @nawendt 

I'm an R-tool developer working in BSC-CNS (Barcelona Supercomputing Center.) We're trying to use gribr to load the GRIB files but we found some problems.

We downloaded some files, system5 and era5 monthly data, from ECMWF MARS archive. For era5 ones, the gribr functions work well. But for system5 ones, we got errors from grib_get_message().

```r
library(gribr)
g <- grib_open("/path/to/file/tas_20001101.grb")
gm <- grib_get_message(g, 1)

ECCODES ERROR : Wrong size for hourOfEndOfOverallTimeInterval it contains 1 values
Error in grib_get_message(g, 1) : gribr: unable to get long array
GRIB ERROR Passed array is too small
```
We tried to load the data with `pygrib` and it works, so we wonder if it is a problem in gribr. I put the file in the link in case it's needed: https://drive.google.com/file/d/1ANMuGn7Y9vXU_UKD4Zp-w8r0ZoxjFgol/view?usp=share_link

Kind regards,  
An-Chi

**Comment by nawendt** [comment]
Thanks for the example file. I'll take a look when I get some more free time. Need to figure out why that key is being sized inappropriately.

**Comment by anchiho** [comment]
Hi @nawendt 

Thanks for the reply. Some update: we found that the problem is actually in the GRIB file. We used other tools to get that attribute and got the same error. The difference is that gribr::grib_get_message takes all the attributes at once while it's not necessary in pygrib or xarray. 

I'd like to ask if it is possible to only retrieve selected attributes while using grib_get_message or grib_select? I see that the main function is in C but I'm not familiar with it, so I'd like to ask your opinion first. 

Thanks for your time,  
An-Chi
